docs/pages/choose-an-edition/migrate-to-cloud.mdx

```diff
@@ -0,0 +1,324 @@
+If you are using infrastructure-as-code tools to 
+To achieve this:
+- Log in to your existing Teleport Enterprise (self-hosted) Cluster and export a collection of the above-mentioned dynamic resource configuration using the `tctl` administrative tool.  An example is shown below:
+
+```code
+$ tsh login --proxy=teleport.example.com --user=admin@example.com
+$ tctl get roles > roles.yaml 
+$ tctl get users > users.yaml 
+```
+
+- Once you have the resource configuration file from the above, proceed to log in to your Teleport Enterprise (cloud-hosted) tenant with an admin user and create the resources from the exported files:
+
+```code
+$ tsh login --proxy=example.teleport.sh --user=admin@example.com
+$ tctl create -f roles.yaml
+$ tctl create -f user.yaml
+```
+
+For your SSO auth connector, most SSO integrations only work for a single configured endpoint. It is recommended to create a separate SSO connector in your Identity Provider specifically for the Teleport Enterprise (cloud-hosted) endpoint, and configure a new Auth Connector in the Teleport Enterprise (cloud-hosted) tenant.
+
+## Step 3/4. Migrate Teleport services and plugins
+
+To migrate services such as Teleport agents, Machine ID Bots, and plugins, start
+by cataloging the various services you're managing with Teleport. The following
+resources should be considered for migration:
+
+ - Teleport agents 
+ - Machine ID Bots
+ - Access Request plugins
+ - The Teleport Event Handler
+
+Before migrating services, make sure you are logged in to your new cloud-hosted
+Teleport Enterprise account.
+
+You can migrate Teleport services all at once or gradually, depending on your
+business requirements. If running Teleport at scale, you'll generally want to
+use a configuration management tool to automate and streamline the process of
+carrying out the actions involved in migrating agent configurations. 
+
+### Migrate Teleport agents
+
+To migrate Teleport agents:
+
+1. For each agent and Machine ID bot, obtain a valid join token. We recommend
+   using a [delegated join
+   method](https://goteleport.com/docs/reference/join-methods/#delegated-join-methods).
+
+1. If using ephemeral tokens, ensure you specify the appropriate token type to
+   match the Teleport services. Token types can include `node`, `app`, `kube`,
+   `db`, `windowsdesktop` and others depending on the service you wish to join
+   to your Teleport Cluster.
+
+1. In the following example, a new token is created with the TTL of fifteen
+   minutes:
+
+   ```code
+   $ tctl tokens add --type node,app,db --ttl 15m
+   ```
+
+   In this command, we assigned the token the `node`, `app` and `db` type,
+   indicating that it will allow an agent to join which is running the Teleport
+   `ssh_service`, `db_service` and `app_service`.
+
+   Copy the token so you can use it later in this guide.
+
+1. Stop Teleport services on the agent (if applicable).
+
+1. Update the `proxy_server` or `auth_servers` field in the agent configuration

```

Comment:
Might be worth just mentioning `proxy_server` here, with a note saying "If you don't have an `auth_server` or `auth_servers` entry rather than a `proxy_server` entry, you should migrate your config to `version: v3` and switch to using `proxy_server` as in the example)

Folks might get confused between `auth_server` (string used in config version `v3`) and `auth_servers` (list used in config version `v1/v2`) and break connectivity.